planner/core/plan.go

```diff
@@ -177,7 +177,10 @@ type LogicalPlan interface {
 	PreparePossibleProperties(schema *expression.Schema, childrenProperties ...[][]*expression.Column) [][]*expression.Column
 
 	// exhaustPhysicalPlans generates all possible plans that can match the required property.
-	exhaustPhysicalPlans(*property.PhysicalProperty) []PhysicalPlan
+	// It will return:
+	// 1. All possible plans that can match the required property.
+	// 2. Whether the SQL hint can satisfy the required property. Return true if there is no hint.
+	exhaustPhysicalPlans(*property.PhysicalProperty) (physicalPlans []PhysicalPlan, hintFitsProp bool)
```

Comment:
Does `hintFitsProp` is true means all the candidate physical plan in the return result satisfy the hint?

Reply:
Yes